Clickstream/EventProcessor/Core/EventProcessor.swift

```diff
@@ -33,6 +33,16 @@ final class DefaultEventProcessor: EventProcessor {
     
     func createEvent(event: ClickstreamEvent) {
         self.serialQueue.async { [weak self] in guard let checkedSelf = self else { return }
+            #if EVENT_VISUALIZER_ENABLED
+            /// Sent event data to client with state received
+            /// to check if the delegate is connected, if not no event should be sent to client
+            if let stateViewer = Clickstream._stateViewer {
+                /// creating the EventData object and setting the status to received.
+                let eventsData = EventData(msg: event.message, state: .received)
+                /// Sending the eventData object to client
+                stateViewer.sendEvent(eventsData)
```

Comment:
when we send an event to stateviewer, would internally it store to persistence? or just store on memcache?

Reply:
No it would not persist the data. I would store the events in memory for that life cycle.